File: vue.jsonforms/src/entities/firm-details/LocumCompanyDetails.ts
import { ContactNumber } from "@/entities/ContactNumber";

export class LocumCompanyDetails {
  public firmName?: string;
  public companyNumber?: string;
  public fcaStatus?: string;
  public companyHouseStatus?: string;
  public firmReferenceNumber?: string;
  public registeredAddress?: string;
  public isRegisteredAddressChanged: boolean = false;
  public tradingAddress?: string;
  public isTradingAddressChanged: boolean = false;
  public isTradingSameAsRegisteredAddress: boolean = false;
  public emailAddress?: string;
  public contactNumber?: ContactNumber;
  public website?: string;
  public representativeTitle?: string;
  public representativeForename?: string;
  public representativeSurname?: string;
  public representativeEmailAddress?: string;
  public representativeMobileNumber?: ContactNumber;
  public representativeRole?: string;
}

File: CSharp.Functions/Common/Entities/FirmDetails/Locum/LocumCompanyDetails.cs
namespace Common.Entities
{
    public class LocumCompanyDetails
    {
        public string? FirmName { get; set; }
        public string? CompanyNumber { get; set; }
        public string? FirmReferenceNumber { get; set; }
        public string? RegisteredAddress { get; set; }
        public bool IsRegisteredAddressChanged { get; set; }
        public string? TradingAddress { get; set; }
        public bool IsTradingAddressChanged { get; set; }
        public bool IsTradingSameAsRegisteredAddress { get; set; }
        public string? EmailAddress { get; set; }
        public ContactNumber? ContactNumber { get; set; }
        public string? Website { get; set; }
        public string? RepresentativeTitle { get; set; }
        public string? RepresentativeForename { get; set; }
        public string? RepresentativeSurname { get; set; }
        public string? RepresentativeEmailAddress { get; set; }
        public ContactNumber? RepresentativeMobileNumber { get; set; }
        public string? RepresentativeRole { get; set; }
    }
}
